src/DependencyInjection/DoctrineAdapterExtension.php

```
@@ -29,6 +53,193 @@ public function load(array $configs, ContainerBuilder $container)
+     * @param string $class
+     * @param string $name
+     * @param array $provider
+     */
+    private function createDoctrineCacheDefinition(ContainerBuilder $container, $typeServiceId, $class, $name, array $provider)
+    {
+        $namespace = is_null($provider['namespace']) ? $name : $provider['namespace'];
+
+        // Create a service for the requested doctrine cache
+        $definition = new Definition($class);
+        $definition->addMethodCall('setNamespace', [$namespace])
+            ->setPublic(false);
+
+        $type = $provider['type'];
+        switch ($type) {
+            case 'memcache':
+            case 'memcached':
+            case 'redis':
+                if (!empty($provider['id'])) {
+                    $providerHelperServiceId = $provider['id'];
+                } else {
+                    // Create a new cache provider if none is defined
+                    $providerHelperServiceId = sprintf('cache_adapter_doctrine.provider.%s.helper', $name);
+                    $providerHelperDefinition = $this->createProviderHelperDefinition($type, $provider);
+                    $container->setDefinition($providerHelperServiceId, $providerHelperDefinition);
+                }
+
+                $definition->addMethodCall(sprintf('set%s', ucwords($type)), [new Reference($providerHelperServiceId)]);
+                break;
+            case 'file_system':
+            case 'php_file':
+                $directory = '%kernel.cache_dir%/doctrine/cache';
+                if (null !== $provider['directory']) {
+                    $directory = $provider['directory'];
+                }
+                $extension = is_null($provider['extension']) ? null : $provider['extension'];
+                $definition->setArguments([$directory, $extension]);
+
+                break;
+            case 'mongo':
+            case 'sqlite3':
+            case 'sqlite':
+            case 'riak':
+            case 'chain':
+                break;
```

IIRC these are meant to fail, until we implement them. Should throw an exception instead of breaking.